app/helpers/katello/katello_urls_helper.rb

```diff
@@ -1,15 +1,27 @@
 require 'uri'
+
 module Katello
   module KatelloUrlsHelper
+    def foreman_settings_url
+      Setting[:foreman_url].sub(/^https/, 'http')
+    end
+
     def subscription_manager_configuration_url(host = nil, rpm = true)
       prefix = if host && host.content_source
-                 "http://#{@host.content_source.hostname}"
+                 "http://#{host.content_source.hostname}"
                else
-                 Setting[:foreman_url].sub(/\Ahttps/, 'http')
+                 foreman_settings_url
                end
       config = rpm ? SETTINGS[:katello][:consumer_cert_rpm] : SETTINGS[:katello][:consumer_cert_sh]
 
       "#{prefix}/pub/#{config}"
     end
+
+    def repository_url(content_path, schema = 'http')
+      path = ::Katello::Glue::Pulp::Repos.repo_path_from_host(@host, content_path)
+      return content_path if content_path =~ %r|^([\w\-\+]+)://|
+      return "#{foreman_settings_url}/pulp/repos/#{path}" unless @host.content_source
```

Comment:
To prevent having the `/pulp/repos/#{path}` at two places, I would suggest doing something like:

```ruby
hostname = if @host.content_source
              "#{schema}://#{@host.content_source.hostname}"
           else
              foreman_settings_url(schema)
           end
"#{hostname}/pulp/#{content_type}/#{path}"
```